Sources/FTPropertyWrappers/Keychain - Support/KeychainCommons.swift

```diff
@@ -0,0 +1,168 @@
+    @QueryElement(key: kSecAttrIsNegative) public var isNegative: Bool?
+    @QueryElement(key: kSecAttrAccount) public var account: String?
+    @QueryElement(key: kSecAttrSynchronizable) public var synchronizable: Bool?
+    
+    @QueryElement(key: kSecAttrAccessible) private var _raw_accesible: CFString?
+    public var accesible: AccesibleOption? {
+        get { _raw_accesible.flatMap(AccesibleOption.init(rawValue:)) }
+        set { _raw_accesible = newValue?.rawValue }
+    }
+    
+    @QueryElement(readOnlyKey: kSecAttrCreationDate) public internal(set) var creationDate: Date?
+    @QueryElement(readOnlyKey: kSecAttrModificationDate) public internal(set) var modificationDate: Date?
+
+    // MARK: Override requirements
+
+    open var itemClassIdentity: [String: Any] {
+        get { fatalError("FTPropertyWrappers KeychainItem: error: empty class identity!") }
+    }
+
+    open var itemData: Data {
+        get { fatalError("FTPropertyWrappers KeychainItem: error: empty data!") }
+        set { fatalError("FTPropertyWrappers KeychainItem: error: empty data!") }
+    }
+    
+    // MARK: Query execution support
+    
+    private var itemAttributes: [String: Any] {
+        var attributes = [String: Any]()
+
+        var willUnset: Set<String> = []
+        var conditionalUnset: [(ifPresent: String, unset: String)] = []
+
+        Mirror(reflecting: self).forEachChildInClassHiearchy { child in
+            guard let element = child.value as? ConfiguringElement else {
+                return
+            }
+            element.insertParameters(into: &attributes)
+            element.constraints.forEach { constraint in
+                switch constraint {
+                case .overridenBy(let attribute):
+                    conditionalUnset += [(element.key, attribute as String)]
+                case .override(let attribute):
+                    willUnset.insert(attribute as String)
+                }
+            }
+        }
+
+        conditionalUnset.compactMap { attributes[$0.ifPresent] != nil ? $0.unset : nil }.forEach { willUnset.insert($0) }
+        willUnset.forEach { attributes.removeValue(forKey: $0) }
+
+        return attributes
+    }
+
+    private func configure(from searchResult: [String: Any]) {
+        Mirror(reflecting: self).forEachChildInClassHiearchy { child in
+            (child.value as? ConfiguringElement)?.readParameters(from: searchResult)
+        }
+
+        if let data = searchResult[kSecValueData as String] as? Data {
+            itemData = data
+        } else {
+            print("FTPropertyWrappers KeychainItem insertQuery: warning: update request without data")
+        }
+    }
+
+    private var insertQuery: [String: Any] {
+        var query = itemClassIdentity.merging(itemAttributes) { lhs, rhs in
+            print("FTPropertyWrappers KeychainItem insertQuery: notice: collision found at instance \(self) between \(lhs) and \(rhs)")
```

Comment:
Is this programmer error or does it event matter if you override the key? I would either use assert or no logging.